doorstop/core/reference_finder.py

```diff
@@ -89,7 +89,7 @@ def find_file_reference(ref_path, root, tree, item_path, keyword=None):
                     continue
 
                 log.debug("searching for ref '{}'...".format(keyword))
-                pattern = r"(\b|\W){}(\b|\W)".format(re.escape(keyword))
+                pattern = r"(\b|\W){}(\b|\W)".format(keyword)
```

Comment:
After looking at this piece of code I think we should keep the `keyword` attribute as is. We should add a `regex` attribute which is used as is as a regular expression without these automatic prefix and postfix additions.

Reply:
This sounds good to me. I also was not sure about mixing the existing behavior.